src/DonationForms/ServiceProvider.php

```diff
@@ -80,6 +89,114 @@ public function boot()
             RemoveDuplicateMeta::class,
             UpdateDonationLevelsSchema::class,
         ]);
+
+        $this->registerAsyncData();
+    }
+
+    /**
+     * @unreleased
+     */
+    private function registerAsyncData()
+    {
+        // Async ajax request
+        Hooks::addAction('wp_ajax_givewp_get_form_async_data_for_list_view', GetAsyncFormDataForListView::class);
+        Hooks::addAction('wp_ajax_nopriv_givewp_get_form_async_data_for_list_view', GetAsyncFormDataForListView::class);
+
+        // Filter from give_goal_progress_stats() function which is used by the admin form list views and form grid view
+        Hooks::addFilter('give_goal_progress_stats', GiveGoalProgressStats::class,
+            'maybeChangeGoalProgressStatsActualValue', 999,
+            2);
+
+        // Form Grid
+        add_filter('give_form_grid_goal_progress_stats_before', function () {
+            $usePlaceholder = give(FormGridView::class)->maybeUsePlaceholderOnGoalAmountRaised();
+
+            if ($usePlaceholder) {
+                // Load assets on form grid pages
+                Hooks::addAction('wp_enqueue_scripts', LoadAsyncDataAssets::class);
+
+                //Enable placeholder on the give_goal_progress_stats() function
+                add_filter('give_goal_progress_stats', function ($stats) {
+                    $stats['actual'] = AsyncDataHelpers::getSkeletonPlaceholder('1rem');
+
+                    return $stats;
+                });
+                add_filter('give_goal_shortcode_stats', function ($stats) {
+                    $stats['income'] = 0;
+
+                    return $stats;
+                });
+            }
+        });
+        Hooks::addFilter('give_form_grid_progress_bar_amount_raised_value', FormGridView::class, 'maybeSetProgressBarAmountRaisedAsync',10,2);
+        Hooks::addFilter('give_form_grid_progress_bar_donations_count_value', FormGridView::class, 'maybeSetProgressBarDonationsCountAsync',10,2);
+
+        // Load assets on the admin list form pages and pages with the WordPress clock editor enable
+        $isAdminListFormPage = isset($_GET['post_type']) && 'give_forms' === $_GET['post_type'];
```

Comment:
Looking at the methods in `Give\DonationForms\V2\DonationFormsAdminPage` like `::isShowing` we will need to account for this because our menu items have two links:
```
isset($_GET['page']) && $_GET['page'] === 'give-forms'
```

It looks like they both share the page === give-forms

Reply:
Note: this is referring to clicking on the top level "Donations" item vs "All forms"

<img width="1503" alt="Screenshot 2024-08-21 at 1 54 10 PM" src="https://github.com/user-attachments/assets/9624e22f-4e30-4b45-a9c3-f8770627c3bc">

Reply:
I made a few changes to make sure the assets will be loaded only when necessary. Resolve in 74640b1f13df4db712f1a4786343c5de11f7d299 and 3cd5778205dc34ffb6b91d1c718e0e3cd96dd1d3